### DISCUSSION TITLE
Feature request: Open the Listbox upward if in lower half of window

### DISCUSSION THREAD

**Discussion by therealgilles** [tailwindlabs/headlessui#1688] [Ideas]
It would be great if there was an option to open the Listbox upward when it's in the lower half of the window.

**Comment by RobinMalfait** [comment]
Hey! Thank you for your suggestion!
Much appreciated! 🙏

Headless UI is headless which means that you are in charge of rendering. You can combine Headless UI with tools like Floating UI or Popper.js to achieve this effect.

**Reply by therealgilles** [reply]
Thanks for the prompt reply.

I have a solution but it has two issues:
1) I constantly need to keep track of the position of the Listbox (using event listeners on resize and scroll), when ideally the direction would only be calculated on the `open` event. The onClick handler is controlled internally so not sure how to do it.
2) The default behavior is to show the selected entry at the bottom, and it would be better to show it at the top when rendering upward.

Let me know if you have suggestions to deal with these issues.
